control/config_proxy.go

```diff
@@ -0,0 +1,104 @@
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2016 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package control
+
+import (
+	"encoding/json"
+
+	"golang.org/x/net/context"
+
+	"github.com/intelsdi-x/snap/control/rpc"
+	"github.com/intelsdi-x/snap/core"
+	"github.com/intelsdi-x/snap/core/cdata"
+	"github.com/intelsdi-x/snap/internal/common"
+)
+
+type configProxy struct {
+	cfg *Config
+}
+
+func (cfg *configProxy) GetPluginConfigDataNode(ctx context.Context, r *rpc.ConfigDataNodeRequest) (*rpc.ConfigDataNode, error) {
+	reply := &rpc.ConfigDataNode{}
+	Type := int(r.PluginType)
+	Name := r.Name
+	Version := int(r.Version)
+	result_node := cfg.cfg.GetPluginConfigDataNode(core.PluginType(Type), Name, Version)
+	reply.Node, _ = nodeToJSON(&result_node)
```

Comment:
Thanks for catching that. It is probably better to just not use JSON as an intermediary since it would remove the chance for these errors.

The reason for using JSON was that cdata.ConfigDataNode has a map[string]ConfigValue. ConfigValue is an interface which makes it harder to represent as a protobuf message. I can represent it as a protobuf message instead and provide helper functions to go from node->proto->node to make it simple to use. I think this would end up being better and should probably live in the common proto so it can be re-used.